File: app/src/main/java/com/pioneer/aaron/dolly/IBasePresenter.java
package com.pioneer.aaron.dolly;

import android.app.Activity;
import android.content.Context;

/**
 * Created by Aaron on 4/29/17.
 */

public interface IBasePresenter {
    boolean checkPermissions(Activity activity);

    void loadResInBackground(Context context);

    void toastForkTask(Context context);

    void onDestroy(Context context);
}

File: app/src/main/java/com/pioneer/aaron/dolly/IBasePresenter.kt
package com.pioneer.aaron.dolly

import android.app.Activity
import android.content.Context

/**
 * Created by Aaron on 4/29/17.
 */

interface IBasePresenter {
    fun checkPermissions(activity: Activity): Boolean

    fun loadResInBackground(context: Context)

    fun onDestroy(context: Context)
}
